app/build.gradle

```diff
@@ -48,7 +48,14 @@ String[] archs = ['arm64-v8a', 'armeabi-v7a', 'x86', 'x86_64']
 dependencies {
   // use jdk8 java.time backport, as long app < Build.VERSION_CODES.O
   implementation("com.jakewharton.threetenabp:threetenabp:1.1.1")
-  implementation 'org.kiwix.kiwixlib:kiwixlib:5.2.0'
+
+  // Get kiwixlib online if it is not populated locally
+  if (!file("./libs").exists()) {
```

Comment:
I would get rid of the `!` and flip the branches

Reply:
Maybe, the top branch will be hit 99% of the time though, so I would personally prefer it to be that way round.

Reply:
https://github.com/kiwix/kiwix-build/issues/346#issuecomment-520349486 sounds like we are not doing this anymore.

Reply:
What I am using this for is to test kiwix-lib changes directly. The workflow is:

1. symlink libs directory to the output of kiwix-build
2. Build aar using kiwix-build
3. Build and run the app.
4. Repeat steps 2-4 until you fix whatever bug you are trying to fix.

Turnaround time with this is about a minute.

Reply:
With unit tests turnaround is in seconds if not milliseconds. We should have this discussion [here](https://github.com/kiwix/kiwix-build/issues/346)

Reply:
Sure unit tests would help and might reduce the need for this but we don't have them now. This is a small change that can be implemented now and allows other bugs such as #735 to be solved now. Simply open the effected file and debug using a combination of breakpoints and print statements between c++ recompiles.